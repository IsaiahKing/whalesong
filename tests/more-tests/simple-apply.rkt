#lang planet dyoo/whalesong

(apply + 2 3 4 5 '())
(apply (lambda args args) 2 3 4 5 '())

"now factorial"

(define (f x)
  (cond
   [(apply = `(,x 0))
    1]
   [else
    (apply * `(,x ,(apply f (apply sub1 (apply list x '())) '())))]))


(f 3)
(+ (f 4) (f 5))