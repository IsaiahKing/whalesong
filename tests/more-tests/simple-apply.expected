14
'(2 3 4 5)
"now factorial"
6
144
